migrator/src/test/java/com/google/phonenumbers/migrator/MigrationJobTest.java

```diff
@@ -0,0 +1,110 @@
+package com.google.phonenumbers.migrator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.i18n.phonenumbers.metadata.DigitSequence;
+import com.google.i18n.phonenumbers.metadata.RangeSpecification;
+import com.google.i18n.phonenumbers.metadata.RangeTree;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class MigrationJobTest {
+
+  private static final String TEST_DATA_PATH = "./src/test/java/com/google/phonenumbers/migrator/testing/testData/";
+
+  @Test
+  public void createFromFilePath_invalidPathLocation_expectException() {
+    String fileLocation = "invalid-path-location";
+    try {
+      MigrationFactory.createMigration(Paths.get(fileLocation), "GB");
+      Assert.fail("Expected IOException and did not receive");
+    } catch (IOException e) {
+      assertThat(e).isInstanceOf(NoSuchFileException.class);
+      assertThat(e).hasMessageThat().contains(fileLocation);
+    }
+  }
+
+  @Test
+  public void createFromNumberString_invalidNumberFormat_expectException() {
+    String numberInput = "+44 one2 34 56";
+    String sanitizedNumber = "44one23456";
+    try {
+      MigrationFactory.createMigration(numberInput, "GB");
+      Assert.fail("Expected RuntimeException and did not receive");
+    } catch (RuntimeException e) {
+      assertThat(e).isInstanceOf(IllegalArgumentException.class);
+      assertThat(e).hasMessageThat().contains(sanitizedNumber);
+    } catch (IOException e) {
+      // IOException is not being tested here
+      e.printStackTrace();
+      Assert.fail("Expected RuntimeException and did not receive");
+    }
+  }
+
+  @Test
+  public void createWithCustomRecipes_invalidPathLocation_expectException() {
```

Comment:
You could also test for each create that it is a success and the migratorJob object holds the values in the fields that you expect.